File: 50-powx-n/50-powx-n.cpp
class Solution {
public: 
    
    double pow(double x, int n){
        
        if(n==0) return 1;
        
        double temp;
        
        if(n%2 == 0){
                temp = pow(x, abs(n/2));
                temp = temp*temp;
            }
            else{
                temp = pow(x, (abs(n) - 1)/2);
                temp = x*temp*temp;
            }
        if(n > 0) return temp;
        return 1/temp;
        }
    
    double myPow(double x, int n) {
        return pow(x,n);
    }
};

File: 50-powx-n/50-powx-n.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        
        return self.pow(x, n)
        
    def pow(self, x, n):
                
        if n == 0:
            return 1
        
        if n%2 == 0:
            ans = self.pow(x, abs(n)/2)
            ans *= ans
        else:
            ans = self.pow(x, (abs(n)-1)/2)
            ans = x*ans*ans
            
        return ans if n > 0 else 1/ans
    
    
    